automated/linux/pmqtest/parse_results.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+# SPDX-License-Identifier: MIT
+
+import os
+import re
+
+
+def print_res(res, key, thr):
+    val = res[key]
+    label = 'pass'
+    if key == 'max':
+        if int(val) >= int(thr):
+            label = 'fail'
+    print("t{}-{}-latency {} {} us".format(res['t'], key, label, val))
+
+
+def get_lastlines(filename):
+    # Start reading from the end of the file until ESC is found
+    with open(filename, 'rb') as f:
+        try:
+            f.seek(-2, os.SEEK_END)
+            while f.read(1) != b'\x1b':
+                f.seek(-2, os.SEEK_CUR)
+            return f.readlines()
+        except IOError:
+            # No ESC found
+            f.seek(0, os.SEEK_SET)
+            return f.readlines()
+    return []
+
+
+def parse(filename, thr):
+    fields = ['min', 'avg', 'max']
+
+    rl = re.compile('[ ,:\n]+')
+    rt = re.compile('[ ,#]+')
+    for line in get_lastlines(filename):
+        data = [x.lower() for x in rl.split(line)]
+        res = {}
+        it = iter(data)
+        for e in it:
+            if e in fields:
+                res[e] = next(it)
+
+        if not res:
+            continue
+
+        # The id is constructed from the '#FROM -> #TO' output, e.g.
+        # #1 -> #0, Min    1, Cur    3, Avg    4, Max  119
+        data = rt.split(line)
+        res['t'] = '{}-{}'.format(data[1], data[3])
+
+        print_res(res, 'min', thr)
+        print_res(res, 'avg', thr)
+        print_res(res, 'max', thr)
+
+
+if __name__ == '__main__':
+    import sys
```

Comment:
please move to the top of the file.
Also this looks pretty similar to other 'parse_results.py' files. As suggested by Anders I propose to move them all to automated/lib and merge into single file supporting all outputs. Shouldn't be too much hassle to do that.

Reply:
Most of the rt-tests tools differ more or less in the output *sig*. Only two parse_results.py are identical. Indeed I could add some logic to detect the output format, I haven't really thought about this.

Reply:
IMHO it can be as simple as runtime parameter.  No need for any complicated detection logic. I noticed that the differences are small enough to be just included as a function in single parse_results.py script.

Reply:
Done